lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -1424,20 +1425,15 @@ someEnclosingDeclMatches(SourceRange ReferenceRange,
         return true;
     }
 
-    DC = DC->getParent();
-  } while (DC);
+    DC = D->getDeclContext();
+  } while (true);
```

Comment:
Can this become a `for(;;)` loop? If not, can it at least be a `while(true)` loop instead of `do {} while(true)` so its lack of natural termination is obvious from the start?

Reply:
Oh. Right.